client/engine/engine.go

```diff
@@ -453,6 +521,24 @@ func (e *Engine) attemptProgress(objective protocols.Objective) (outgoing Object
 	return
 }
 
+func (e Engine) registerVirtualChannelWithManagers(vfo virtualfund.Objective) error {
+	prefund := vfo.V.PreFundState()
+	startingBalance := big.NewInt(0)
+	// TODO: Assumes one asset for now
+	startingBalance.Set(prefund.Outcome[0].Allocations[0].Amount)
+
+	switch vfo.MyRole {
```

Comment:
I'm not sure this is correct:

Reply:
I think when you are trying to detect the case where you are Bob, you want to match against `vfo.myRole` being `n`? `n=2` only if there is a single intermediary.

Reply:
Ah good point. I think there's probably some other assumptions that `n=2` in our code but I might as well clean this one up!

Reply:
Done in [0490cd7](https://github.com/statechannels/go-nitro/pull/836/commits/0490cd7480d27e8c67ad81e7f18c9cb7eb093bd5)